### PR TITLE
Fix handling of variable to match spec behaviour.

### LINKED ISSUE #25: Filters not resolving properly?
Hi,

I'm having a bit of an issue with filters right now, which is confusing me.

I have a query which looks something like this:

```
query Foo($bar: String!) {
  system() {
    component(filter: { someKey: [$bar] }) {
     ...
  }
}
```

When I check the filter in the args passed to the resolvers, it's an empty object. If I change the filter to

```
    component(filter: { someKey: $bar }) {
```

I see the correct value in the filter.

Am I doing something wrong, or is there potentially a bug here?

Thanks.

### PR DESCRIPTION
**Pull request by ruiaraujo**
In practise this means that we now properly handle nested variable in objects and lists.

closes #25.

### CODE REVIEW

File: src/__tests__/inspect.test.ts [hunk 1]
```diff
@@ -166,16 +165,37 @@ describe("inspect", () => {
...
+    expect(inspect([[new Foo()]])).toEqual("[[[Foo]]]");
 
     (Foo.prototype as any)[Symbol.toStringTag] = "Bar";
-    expect(inspect([[[new Foo()]]])).toEqual("[[[[Bar]]]]");
+    expect(inspect([[new Foo()]])).toEqual("[[[Bar]]]");
+  });
+});
+
+describe("createInspect", () => {
+  test("can increase the depth", () => {
```
Comment (ruiaraujo) [inline src/__tests__/inspect.test.ts]: This changes were required to allow having deeper inspection on the tests.

Also made the defaults match `graphql-js`.

**Comment by ruiaraujo** [comment]
:+1:

**Comment by ruiaraujo** [comment]
:+1:

**Comment by boopathi** [comment]
👍

**Pull request merged**